### PR TITLE
add other builtins to interstage IO

### PR DESCRIPTION
**Pull request by steve-lunarg**
WIP: add other builtins to interstage IO

This fixes PR #632 so that:

(a) The 4 PerVertex builtins are added to an interface block for all stages except fragment.

(b) Other builtin qualified variables are added as "loose" linkage members.

(c) Arrayness from the PerVertex builtins is moved to the PerVertex block.

(d) Sometimes, two PerVertex blocks are created, one for in, one for out (e.g, for some GS that both reads and writes a Position)

### CODE REVIEW

File: hlsl/hlslParseHelper.cpp [hunk 1]
```diff
@@ -1446,8 +1449,75 @@ void HlslParseContext::addInterstageIoToLinkage()
...
+                }
+                member.type->clearArraySizes();
+            }
+
+            ioBlockTypes[blockId]->push_back(member);
+        }
+    }
+
+    // If there were perVertex items, add the block to the linkage.  Handle in and out separately.
+    for (int blockId=0; blockId<=1; ++blockId) {
```
Comment (johnkslang) [inline hlsl/hlslParseHelper.cpp]: Use spaces around ops, like other loops/etc. do.

File: glslang/Include/Types.h [hunk 2]
```diff
@@ -1333,6 +1342,53 @@ class TType {
         }
     }
 
+    virtual bool isLooseBuiltIn(EShLanguage language) const
+    {
+        // Any stage
```
Comment (johnkslang) [inline glslang/Include/Types.h]: Might be simpler to start:

```
if (language == EShLangFragment || language == EShLangCompute)
    return true;
```

Comment (steve-lunarg) [inline glslang/Include/Types.h]: Agreed, that's better.  Will fix.

**Comment by steve-lunarg** [comment]
Repushed with review changes:

1. Add spaces around for() operators
2. Changed name isLooseBuiltIn -> isLooseAndBuiltIn
3. Rephrased isLooseAndBuiltIn in terms of isPerVertexAndBuiltIn, which is much shorter and more robust against new ones appearing.

**Pull request merged**